simplify `meeting_start_logs` view query (iss. #520)

lsloan opened the pull request:
Resolves #520.

Debugging other problems brought to my attention that the query from migration 29 is too complex.  The queue ID didn't need to be selected as a different column just to be used in the join later.  It seems better to make this simpler so future maintenance will be easier.

lsloan commented:
@jonespm, thanks for the review!

To address your comments…

1. From the discussion in meetings (this week and last week) and comments in #656 (specifically, [this comment](https://github.com/tl-its-umich-edu/remote-office-hours-queue/issues/656#issuecomment-2944905367)) Zhen convinced me that it would be OK to use `DISTINCT` in this view.  That other issue would remain open to determine why lines are repeated in the log.
2. No, this change does **_NOT_** address #667.  While looking into that issue, I noticed that this query was not what I intended.  It was more complex than necessary.
3. I thought it would be nice to put the queue-related columns together in the view.  It turns out that PostgreSQL doesn't support changing column order, even when using `CREATE OR REPLACE`.  The solution is to use a `DROP` first, which is what I do in [lines 12–14 of the migration](https://github.com/lsloan/rohq/blob/e5a107e415232c49322e1ce71d7e6eca57c9721a/src/officehours_api/migrations/0030_update_simpler_meeting_start_logs_view.py#L12-L14).

Re-ordering of the columns is not important.  The migration worked for me, but if it isn't working for you, I can put the queue columns back at the end and revert to a simple `CREATE OR REPLACE` query, without `DROP`.

jonespm commented:
1) Okay as long as nothing is lost here. I'm not 100% convinced that this is the case, but the duplicates I noticed did _seem_ to be duplicates and I wasn't sure the cause. I guess since it operates on the data in the view this should eliminate ones that are distinct in this context.
2) Okay I think this error is related to some code changes then so will still need to figure out a fix or postpone
3) I think the DROP/CREATE is probably more likely to always work. I don't know if we'd need that DROP/CREATE also in the `reverse_sql`. But I also don't know if we'd ever run the reverse through Django, instead doing it manually or through a backup.

lsloan commented:
I've also added the `DROP` to the reverse SQL and removed the `OR REPLACE` from both `CREATE` statements.  It makes no sense to replace if it has been dropped moments before that.

lsloan commented on src/officehours_api/migrations/0030_update_simpler_meeting_start_logs_view.py:
```diff
@@ -0,0 +1,78 @@
+                    parsed_response
+                JOIN officehours_api_queue q ON (response -> 'queue')::int = q.id
+            """,
+            # NOTE: This SQL reverts the view to its state BEFORE migration 0029.
```
OK.  We can do that if you'd like.  My thought is that we'd want to revert to migration 0028 (which is before 0029) because we won't actually use migration 0029.

lsloan commented on src/officehours_api/migrations/0030_update_simpler_meeting_start_logs_view.py:
```diff
@@ -0,0 +1,78 @@
+                CREATE VIEW meeting_start_logs AS
+                WITH parsed_response AS (
+                    SELECT
+                        response::jsonb AS response
```
OK.  As above, if you like that, we can do it.  I was basically skipping migration 0029 because we don't need it.

I'll make those changes and request a new review.

zqian replied:
Great!
